File: backend/backend/src/main/java/com/moovie/backend/data/remote/tmdb/TmdbDataSource.java
package com.moovie.backend.data.remote.tmdb;

import com.moovie.backend.data.remote.tmdb.model.TmdbResponses.*;

import java.util.List;

public interface TmdbDataSource {

    TmdbMovieListResponse getTrendingMovies(int page);

    TmdbMovieDetailResponse getMovieDetail(int movieId);

    TmdbMovieListResponse searchMovies(String query, int page);

    TmdbMovieListResponse discoverMovies(int page, String year, String releaseDateGte, String releaseDateLte,
                                          String sortBy, String withGenres, String withOriginalLanguage,
                                          String withOriginCountry, String voteCountGte);

    TmdbGenreListResponse getGenres();

    List<TmdbCountry> getCountries();

    List<TmdbLanguage> getLanguages();

    TmdbReviewListResponse getMovieReviews(int movieId, int page);
}

File: src/main/kotlin/org/mobyle/data/remote/tmdb/TmdbDataSource.kt
package org.mobyle.data.remote.tmdb

import org.mobyle.data.remote.tmdb.model.*

interface TmdbDataSource {
    suspend fun getTrendingMovies(page: Int): TmdbMovieListResponse
    suspend fun getMovieDetail(movieId: Int): TmdbMovieDetailResponse
    suspend fun searchMovies(query: String, page: Int, year: Int? = null): TmdbMovieListResponse
    suspend fun discoverMovies(
        page: Int,
        year: Int?,
        releaseDateGte: String?,
        releaseDateLte: String?,
        sortBy: String?,
        genres: String?,
        language: String?,
        country: String?,
        voteCountGte: Int?
    ): TmdbMovieListResponse
    suspend fun getGenres(): TmdbGenreListResponse
    suspend fun getCountries(): List<TmdbCountry>
    suspend fun getLanguages(): List<TmdbLanguage>
    suspend fun getMovieReviews(movieId: Int, page: Int): TmdbReviewListResponse
}
